plugins/modules/mount.py

```diff
@@ -792,6 +803,16 @@ def main():
     elif state == 'mounted':
         dirs_created = []
         if not os.path.exists(name) and not module.check_mode:
+            old_umask = None
+            if umask is not None:
+                if not isinstance(umask, int):
+                    try:
+                        umask = int(umask, 8)
+                    except ValueError as e:
+                        module.fail_json(msg="umask must be an octal integer: %s" % (to_native(e)))
+                old_umask = os.umask(umask)
+                os.umask(umask)
```

Comment:
what's the purpose of calling `os.umask(umask)` twice ?

Reply:
Sorry, it's my mistake.